### PR TITLE
Issue #908: setting the preferred min parallelism to be not more than max parallelism

### PR DESCRIPTION
**Pull request by suryasoma**
https://github.com/GoogleCloudDataproc/spark-bigquery-connector/issues/908

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 * GAX has been upgraded to version 2.23.0
 * gRPC has been upgraded to version 1.53.0
 * Netty has been upgraded to version 4.1.89.Final
+* PR #915: Fixing the Issue #908
```
Comment (davidrabinowitz) [inline CHANGES.md:17]: Please change to `Issue #908: ...`

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public ReadSessionResponse create(
                         .setTable(tablePath)
                         .build())
                 .setMaxStreamCount(maxStreamCount)
-                .setPreferredMinStreamCount(preferredMinStreamCount)
+                .setPreferredMinStreamCount(Math.min(preferredMinStreamCount, maxStreamCount))
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java:147]: We need to at least warn the user - please add an `if` prior to the call, and add a warning log that the preferredMinStreamCount is larger than the max

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please rebase the branch to latest master (post 0.29.0)

File: CHANGES.md (hunk 1 above)
Comment (suryasoma) [reply CHANGES.md]: Addressed

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java (hunk 2 above)
Comment (suryasoma) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java]: Addressed.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java [hunk 3]
```diff
@@ -130,6 +130,13 @@ public ReadSessionResponse create(
                   log.debug("using default max parallelism [{}]", defaultMaxStreamCount);
                   return defaultMaxStreamCount;
                 });
+    int minStreamCount = preferredMinStreamCount;
+    if (minStreamCount > maxStreamCount) {
+      minStreamCount = maxStreamCount;
+      log.debug(
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java:136]: This should be a warning log, not debug

File: CHANGES.md [hunk 4]
```diff
@@ -16,6 +16,7 @@
 * GAX has been upgraded to version 2.23.0
 * gRPC has been upgraded to version 1.53.0
 * Netty has been upgraded to version 4.1.89.Final
+* Issue #908: Fixing the `preferred_min_stream_count` must be less than or equal to `max_stream_count`
```
Comment (davidrabinowitz) [inline CHANGES.md:19]: Nit: 
```
Issue #908: Making sure that `preferred_min_stream_count` is less than or equal to `max_stream_count`
```

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**